samples/query_external_gcs_temporary_table.py

```diff
@@ -30,7 +31,7 @@ def query_external_gcs_temporary_table():
         bigquery.SchemaField("name", "STRING"),
         bigquery.SchemaField("post_abbr", "STRING"),
     ]
-    external_config.options.skip_leading_rows = 1
+    typing.cast(bigquery.CSVOptions, external_config.options).skip_leading_rows = 1
```

Comment:
We now have a [`csv_options` property](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.external_config.ExternalConfig.html#google.cloud.bigquery.external_config.ExternalConfig.csv_options), where this type cast would be unnecessary.

```suggestion
    external_config.csv_options.skip_leading_rows = 1
```

Looks like we are careful to ensure that sub-setters are possible, since we use the same `_properties` reference. https://github.com/googleapis/python-bigquery/blob/44221f437f93fae2fac10d074b39a28008b9ad01/google/cloud/bigquery/external_config.py#L891

Edit: We might still need something since the type is annotated as `Optional`. I wonder if we should remove the `if self.source_format == ....` logic and always do `setdefault` ? It means we don't have client-side protections against setting values for multiple source formats, but I'm okay letting the server-side do that validation.

Reply:
How about sticking in an explicit assertion `assert foo is not None`? Mypy understands that and we can get rid of casting the optional values, which cannot be `None` in a particular runtime context.